wright/src/grammar/parsers/expression/block.rs

```diff
@@ -1,12 +1,41 @@
-use crate::grammar::ast::{eq::AstEq, Block, Expression};
+use crate::grammar::ast::{eq::AstEq, Block, Expression, Statement};
 use crate::grammar::model::{Fragment, HasFragment};
 use crate::grammar::parsers::expression::ToExpression;
+use crate::grammar::parsers::whitespace::token_delimiter;
+use crate::grammar::parsers::with_input;
+use nom::bytes::complete::tag;
+use nom::combinator::{map, opt};
+use nom::multi::many0;
+use nom::sequence::{delimited, pair, terminated};
 use nom::IResult;
 
 impl<'s> Block<'s> {
+    /// Block delimiters. Should not change.
+    pub const DELIMITERS: (&'static str, &'static str) = ("{", "}");
+
+    fn inner(
+        input: Fragment<'s>,
+    ) -> IResult<Fragment<'s>, (Vec<Statement<'s>>, Option<Box<Expression<'s>>>)> {
+        pair(
+            many0(terminated(Statement::parse, token_delimiter)),
+            opt(map(Expression::parse, |e| Box::new(e))),
+        )(input)
```

Comment:
Is there any syntax that correctly parses as both a statement and expression? (A conditional expression maybe?) If so, the result expression will incorrectly be `None`.

Reply:
There shouldn't be. Conditionals are always expressions